examples/language_model/bert/static/run_glue_with_sparaity.py

```diff
@@ -0,0 +1,456 @@
+            collate_fn=batchify_fn,
+            feed_list=[input_ids, token_type_ids, labels],
+            num_workers=0,
+            return_list=False)
+        dev_batch_sampler_mismatched = paddle.io.BatchSampler(
+            dev_ds_mismatched, batch_size=args.batch_size, shuffle=False)
+        dev_data_loader_mismatched = DataLoader(
+            dataset=dev_ds_mismatched,
+            batch_sampler=dev_batch_sampler_mismatched,
+            collate_fn=batchify_fn,
+            num_workers=0,
+            feed_list=[input_ids, token_type_ids, labels],
+            return_list=False)
+    else:
+        dev_ds = load_dataset('glue', args.task_name, splits='dev')
+        dev_ds = dev_ds.map(trans_func, lazy=True)
+        dev_batch_sampler = paddle.io.BatchSampler(
+            dev_ds, batch_size=args.batch_size, shuffle=False)
+        dev_data_loader = DataLoader(
+            dataset=dev_ds,
+            batch_sampler=dev_batch_sampler,
+            collate_fn=batchify_fn,
+            num_workers=0,
+            feed_list=[input_ids, token_type_ids, labels],
+            return_list=False)
+
+    # Create the training-forward program, and clone it for the validation
+    with paddle.static.program_guard(main_program, startup_program):
+        num_class = 1 if train_ds.label_list is None else len(
+            train_ds.label_list)
+        model, pretrained_state_dict = model_class.from_pretrained(
+            args.model_name_or_path, num_classes=num_class)
+        loss_fct = paddle.nn.loss.CrossEntropyLoss(
+        ) if train_ds.label_list else paddle.nn.loss.MSELoss()
+        logits = model(input_ids, token_type_ids)
+        loss = loss_fct(logits, labels)
+        dev_program = main_program.clone(for_test=True)
+
+    # Create the training-backward program, this pass will not be
+    # executed in the validation
+    num_training_steps = args.max_steps if args.max_steps > 0 else len(
+        train_data_loader) * args.num_train_epochs
+    with paddle.static.program_guard(main_program, startup_program):
+        lr_scheduler = LinearDecayWithWarmup(
+            args.learning_rate, num_training_steps, args.warmup_steps)
+        # Generate parameter names needed to perform weight decay.
+        # All bias and LayerNorm parameters are excluded.
+        decay_params = [
+            p.name for n, p in model.named_parameters()
+            if not any(nd in n for nd in ["bias", "norm"])
+        ]
+        optimizer = paddle.optimizer.AdamW(
+            learning_rate=lr_scheduler,
+            epsilon=args.adam_epsilon,
+            parameters=model.parameters(),
+            weight_decay=args.weight_decay,
+            apply_decay_param_fun=lambda x: x in decay_params)
+
+        # Keep Pooler and task-specific layer dense.
+        # Please note, excluded_layers must be set before calling `optimizer.minimize()`.
+        sparsity.set_excluded_layers(main_program, ['linear_72', 'linear_73',])
```

Comment:
How about using `model.bert.pooler.dense.full_name(), model.classifier.full_name()`  to replace `'linear_72', 'linear_73'`, which might be more meaningful.

Reply:
It's a great suggestion. Updated.